library/network/src/modules/SuSEFirewall.rb

```diff
@@ -805,11 +830,67 @@ def Write
       true
     end
 
+    # Function returns if the interface is in zone.
+    #
+    # @param [String] interface
+    # @param	string firewall zone
+    # @return	[Boolean] is in zone
+    #
+    # @example IsInterfaceInZone ("eth-id-01:11:DA:9C:8A:2F", "INT") -> false
+    def IsInterfaceInZone(interface, zone)
+      interfaces = @SETTINGS[zone][:interfaces]
+      interfaces.include?(interface)
+    end
+
+    # Function returns the firewall zone of interface, nil if no zone includes
+    # the interface. Firewalld does not allow an interface to be in more than
+    # one zone, so no error detection for this case is needed.
+    #
+    # @param string interface
+    # @return string zone, or nil
+    def GetZoneOfInterface(interface)
+      Builtins.foreach(GetKnownFirewallZones()) do |zone|
+        return zone if IsInterfaceInZone(interface, zone)
+      end
+
+      nil
+    end
+
+    # Function returns name of the zone. Since firewalld zones are move verbose,
+    # only 'dmz' is converted to a fuller name. Please note that as of yet, no
+    # translation is possible.
```

Comment:
Why no translations?

IMHO we should take the nine firewalld zones from ["Which zones are available?"](https://twoerner.fedorapeople.org/firewalld/doc/firewalld.zones.html) and make a list of nine translatable strings, just like [the original code does it](https://github.com/yast/yast-yast2/blob/9c7caad3f3332f607937e9b236d90c202e4b1de9/library/network/src/modules/SuSEFirewall.rb#L114). That is simple and correct, while this code seems neither. Am I missing something?